internal/model/model.go

```diff
@@ -42,6 +101,19 @@ func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 				m.Page--
 			}
 		}
+
+	case fileWatchMsg:
+		newFileInfo, err := os.Stat(m.FileName)
+		if err == nil {
+			if newFileInfo.ModTime() != fileInfo.ModTime() {
+				fileInfo = newFileInfo
+				_ = m.Load()
+				if m.Page >= len(m.Slides) {
+					m.Page = len(m.Slides) - 1
+				}
+			}
+		}
```

Comment:
Can you fix the code to have a more flat structure, something like this?

```suggestion
		if err != nil {
			return m, fileWatchCmd()
		}
		// File unchanged, skip reload
	        if newFileInfo.ModTime() == fileInfo.ModTime() {
			return m, fileWatchCmd()
	        }
		fileInfo = newFileInfo
		_ = m.Load()
		if m.Page >= len(m.Slides) {
			m.Page = len(m.Slides) - 1
		}
```

Reply:
I don't think that repeating the line `return m, fileWatchCmd()` 3 times is a better option :)
Anyway, I reduce the level of nesting a little bit.

Reply:
Thanks!